src/http/web_socket/protocol.cr

```diff
@@ -320,6 +320,9 @@ class HTTP::WebSocket::Protocol
 
     if (host = uri.hostname) && (path = uri.request_target)
       tls = uri.scheme.in?("https", "wss")
+      if (user = uri.user) && (password = uri.password)
+        headers["Authorization"] = "Basic #{Base64.strict_encode("#{user}:#{password}")}"
```

Comment:
This probably should not override an existing `Authorization` header. WDYT?
```suggestion
        headers["Authorization"] ||= "Basic #{Base64.strict_encode("#{user}:#{password}")}"
```

Reply:
I'd be happy with it either way - I can see reasonable arguments on either side. To be honest, throwing an exception is probably the safest (but maybe annoying in practice).

Not clobbering a custom header makes sense - if I was explicitly setting the header I can imagine being very frustrated at it being overridden, and probably less frustrated at the inverse.